subiquity/server/controllers/identity.py

```diff
@@ -135,3 +158,49 @@ async def validate_username_GET(self, username: str) -> UsernameValidation:
             return UsernameValidation.TOO_LONG
 
         return UsernameValidation.OK
+
+    # The identity section (and user-data) section are entirely optional
+    # on Desktop, but setting the variant doesn't happen until controllers
+    # are started. For the identity controller to properly validate
+    # this behavior, we need to do look-ahead on the source controller
+    # logic for loading the available sources.
+    # See: https://github.com/canonical/subiquity/pull/1965
+    def _install_source_is_desktop(self) -> bool:
+        # Check the autoinstall config first
+        source_config = self.app.autoinstall_config.get("source")
+        if (
+            source_config is not None
+            and (id := source_config.get("id")) is not None
+            and "server" not in id
```

Comment:
I'm wary of checking the string set on the id, it's semi free form.
What's the motivation for this section?  It looks like we can drop it and get the same outcome?  If I'm missing something let me know.

Reply:
in `subiquity/server/controllers/source.py` in `load_autoinstall_data` we get the source id with a `.get("id")` on the autoinstall data and do an exact match on the IDs in the catalog via `get_matching_source` in `subiquity/models/source.py` so I felt that it wasn't that bad of a comparison to just check that `server` wasn't in the name. 

In the autoinstall case if the id the user provides doesn't match any sources in the catalog then we get a `KeyError` anyways so I wasn't super concerned with weird outcomes if the user provides something unrealistic.

Reply:
*but this section matches the source controller/model logic of letting autoinstall inform us of the source id